types/src/blob.rs

```diff
@@ -9,6 +9,15 @@ use crate::consts::appconsts;
 use crate::nmt::Namespace;
 use crate::{bail_validation, Error, Result};
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct SubmitOptions {
+    // TODO: giving negative fees always results in insufficient fee err, but celestia
+    //       have it like so, should we do a u64 there?
```

Comment:
I see `int64` there, is celestia able to correctly reject `i64::MAX..u64::MAX` values? If so I don't think there's harm to it

Reply:
Actually in other places we used `Uint` for fees and it was serialized as numeric strings. What celestia does here? I prefer `u64` if we can not use `Uint` and have a `serialize_with` that checks if the number is bigger than `i64`. Or maybe open a bug to them.

Reply:
actually having `u64` at let the server return an error is more acceptable for me than `i64`.

Reply:
unfortunately requests with value in this range just fails, I'll stick to the exact types celestia is using and delete this todo